docs/src/dev_guide/auto_generation.md

```diff
@@ -25,10 +25,45 @@ reasons to consider using this approach:
 InfrastructureSystems.generate_structs("./src/descriptors/power_system_structs.json", "./src/models/generated")
 ```
 
+## Struct Descriptor Rules
+Each struct descriptor must define the following fields:
+
+- ``struct_name``: Name of struct
+- ``docstring``: Struct docstring
+- ``fields``: Array of struct members. See below for requirements.
+- ``supertype``: Declare the struct with this parent type.
+
+Required fields for each struct member:
+
+- ``name``: Name of field
+- ``data_type``: Type of field
+
+Optional fields for each struct member:
+
+- ``accessor_module``: Set this if the the getter/setter function are
+reimplementing a method defined in a different module.
+- ``comment``: Field comment
+- ``default``: The constructors will define this as a default value.
+- ``exclude_setter``: Do not generate a setter function for this field.
+- ``internal_default``: Set to true for non-user-facing fields like
+InfrastructureSystemsInternal that have default values.
+- ``needs_conversion``: Set to true if the getter function needs apply unit
+conversion. The type must implement ``get_value(::InfrastructureSystemsComponent,
+::Type) for this combination of component type and member type.``
+- ``null_value``: Value to indicate the value is null, such as 0.0 for floats.
+If all members in the struct define this field then a "demo" constructor will be
+generated. This allows you to enter ``val = MyType(nothing)`` in the REPL and
+see the layout of a struct without worrying about valid values.
+- ``valid_range``: Define this as a Dict with ``min`` and ``max`` and
```

Comment:
Can we add a line to recommend use NULL when the limit doesn't apply. For instance when there is not max.